Assets/Scripts/StandardSamples/UI/MatchMaking/UISessionEntry.cs

```diff
@@ -84,13 +102,21 @@ public void EnableButtonsBySessionState(bool updating, OnlineSessionState state)
             {
                 NameTxt.transform.gameObject.SetActive(true);
 
-                StartButton.interactable = !updating && (state == OnlineSessionState.Pending || state == OnlineSessionState.Ended);
-                EndButton.interactable = !updating && state == OnlineSessionState.InProgress;
-                LeaveButton.interactable = true;
+                if (isOwner)
+                {
+                    StartButton.interactable = !updating && (state == OnlineSessionState.Pending || state == OnlineSessionState.Ended);
+                    EndButton.interactable = !updating && state == OnlineSessionState.InProgress;
+                    ModifyButton.interactable = !updating;
+                }
+                else
+                {
+                    StartButton.interactable = false;
+                    EndButton.interactable = false;
+                    ModifyButton.interactable = false;
+                }
```

Comment:
I would insert a block of comments first thing within each clause of the conditional explaining in plain language (doesn't have to be super specific - just general) what is happening in each clause and why.

Reply:
Sure, that seems good. This is a seemingly arbitrary list of things doing things.